code/xcvm/cosmwasm/contracts/gateway/src/state.rs

```diff
@@ -1,14 +1,13 @@
 use crate::msg;
 
+use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Addr, IbcEndpoint, StdResult, Storage};
 use cw_storage_plus::{Item, Map};
-use schemars::JsonSchema;
-use serde::{Deserialize, Serialize};
 use xc_core::{AssetId, InterpreterOrigin, NetworkId};
 
 pub(crate) type ChannelId = String;
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[cw_serde]
```

Comment:
what is problem with this code. 
as soon as will use Substrate(sp) for doing arbitrary dispatch calls to runtime, or depending on XCM, which is not Scheam (but burried some in Polkadot repo). I wouill not be able to use these types with Schema in same crate. 

Our Polkadot grand tells need for that dep.  Also it hurdens mobing shared code to core. Sure this exact line is not an issue so.

Also, does Schema increase wasm side or it compiled away? Is that metadata stored with code?

Reply:
We’re using forked cosmwasm-schema with no_std support though so compatibility with no_std environments shouldn’t be an issue?

Reply:
@dzmitry-lahoda, so I’ve spent some time trying to check this but at the moment the contracts don’t compile in no_std environment (I’ve tried building for thumbv6m-none-eabi target).  I’ve looked into this but unless we need to be able to compile the contracts for no_std right now, I’d say let’s go with this PR and then figure out no_std later.  For example, I have some PRs for cosmwasm to support no_std (a bit different approach to your fork) so this should be doable in the future.  For JsonSchema I’m pretty sure that the solution is to have cosmwasm-std not derive it if building no_std.

Reply:
tbh whater no std works consiatently accross cw and ibc and substrate, good.

without std feature and wasm32 target work on my pr.
sure more strict version of embeeded target not.

Reply:
i make all jsonschema conditional in all places and it works consistently ok.

Reply:
> without std feature and wasm32 target work on my pr.

But that doesn’t really means the code works without std.  If our core is no_std but a dependency is not than wasm32 build will work fine but it doesn’t mean the code is no_std.  This PR compiles for wasm32 as well.

> i make all jsonschema conditional in all places and it works consistently ok.

So with cw_serde, this just needs to happen in no_std fork of cosmwasm-std.

Also, what’s the status of your PR? ’Cuz it looks like I’m kind of blocked on it?

Reply:
did you had time to review it? if you are blocked please review. what exactly you arr blocked on?

Reply:
i mean it was 100 green, but because of some changes  needed to rebase. i guess blocking parts where here.

Reply:
> did you had time to review it? if you are blocked please review. what exactly you arr blocked on?

I have approved it already a while back.  I’m actually not clear on the blocking status, but if your PR improves no_std situation than I guess this PR would benefit from that? Then again, this PR does compile for wasm32 so maybe that’s not an issue.

> i mean it was 100 green, but because of some changes needed to rebase.

And this is another reason to prefer small PRs.  It’s easier to rebase them and merge them quickly. :]

Reply:
i replaced xcvm internal, size of pr depends on size of replacement in grrat degree.

Reply:
OK, so what’s the next step with those PRs? As far as I see, this PR doesn’t need any further changes and can be merged.